### DISCUSSION TITLE
Question about Apache-2.0 in space

### DISCUSSION THREAD

**Discussion by tomjmech** [nasa/cFS#881] [General]
Excerpt from the Apache-2.0 license:
... each Contributor hereby grants to You
        a perpetual, worldwide ... copyright license ...
        a perpetual, worldwide ... patent license ...

The copyright and patent grants are restricted: **worldwide**.
A satellite around Earth is arguably not included by that definition.
Or is it? What about software on the Moon or on Mars?

Is Apache-2.0 licensed software legally usable in space?

**Comment by jefferyjohn** [comment]
Obligatory note that I'm not a lawyer, but it could potentially be argued that software distributed from Earth by means of rocket could retain the copyright license.

```Subject to the terms and conditions of this License, each Contributor hereby grants to You a ... copyright license to ... distribute the Work and such Derivative Works in Source or Object form.```

Without means of patent litigation in space (at this time), it could also potentially be argued that the License cannot be revoked on those grounds. 

``` If You institute patent litigation against any entity (including a cross-claim or counterclaim in a lawsuit) ... then any patent licenses granted to You under this License for that Work shall terminate as of the date such litigation is filed.```

However, litigation could be filed (on Earth) for Contributions that are not licensable by their contributors.

```...where such license applies only to those patent claims licensable by such Contributor that are necessarily infringed by their Contribution(s) alone or by combination of their Contribution(s) with the Work to which such Contribution(s) was submitted. ```

TL;DR: If the Apache-2.0 licensed software is legally usable on Earth, it _may_ legally be able to sent into space
